pkgs/applications/graphics/structorizer/default.nix

```diff
@@ -0,0 +1,105 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, jdk11
+, makeDesktopItem
+, makeWrapper
+, copyDesktopItems
+}:
+
+stdenv.mkDerivation rec {
+  pname = "structorizer";
+  version = "3.32-11";
+
+  desktopItems = [
+    (makeDesktopItem {
+      type = "Application";
+      name = "Structorizer";
+      desktopName = "Structorizer";
+      genericName = "Diagram creator";
+      comment = meta.description;
+      icon = pname;
+      exec = pname;
+      terminal = false;
+      mimeTypes = [ "application/nsd" ];
+      categories = [
+        "Development"
+        "Graphics"
+        "VectorGraphics"
+        "RasterGraphics"
+        "ComputerScience"
+      ];
+      keywords = [ "nsd" "diagrams" ];
+    })
+  ];
+
+  src = fetchFromGitHub {
+    owner = "fesch";
+    repo = "Structorizer.Desktop";
+    rev = version;
+    hash = "sha256-rGyeOcGm6uBplgTjMIOy/xRekfHacwDy9kkMigmRSdk=";
+  };
+
+  nativeBuildInputs = [ jdk11 makeWrapper copyDesktopItems ];
+
+  patches = [ ./makeStructorizer.patch ./makeBigJar.patch ];
+
+  postPatch = ''
+    chmod +x makeStructorizer
+    chmod +x makeBigJar
+
+    patchShebangs --build makeStructorizer
+    patchShebangs --build makeBigJar
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+
+    ./makeStructorizer
+    ./makeBigJar
+
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    install -D ${pname}.jar -t $out/share/java/
+    install -d $out/bin
+    makeWrapper ${jdk11}/bin/java $out/bin/${pname} \
+      --add-flags "-jar $out/share/java/${pname}.jar"
+
+    mkdir -p $out/share/mime/application
+    cat << EOF > $out/share/mime/application/nsd.xml
+    <?xml version="1.0" encoding="UTF-8"?>
+    <mime-info xmlns="http://www.freedesktop.org/standards/shared-mime-info">
+      <mime-type type="application/nsd">
+             <comment xml:lang="en">Nassi-Shneiderman diagram</comment>
+             <comment xml:lang="de">Nassi-Shneiderman-Diagramm</comment>
+             ...
+             <magic priority="50">
+                    <match type="string" value='xmlns:nsd="https://structorizer.fisch.lu"'/>
+             </magic>
```

Comment:
`<match>` apparently requires an offset. I'm not sure whether this is even the correct use-case for the `<magic>` element. If you find another way of specifying that the file should contain `xmlns:nsd="https://structorizer.fisch.lu"`, that would be nice, but otherwise I think it's fine to remove it and rely fully on file suffix. That's what I did to get it to work.